File: 프로그래머스/lv1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.py
def solution(numbers):
    numbers.sort()
    ansset = set()
    for i in range(len(numbers)):
        for j in range(len(numbers)):
            if i != j:
                ansset.add(numbers[i] + numbers[j])
    answer = list(ansset)
    answer.sort()
    return answer

File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.cpp
#include <vector>
#include <set>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> numbers) {
    vector<int> answer;
    set<int> seen;
    
    for (int i = 0; i < numbers.size() - 1; i++) {
        for (int j = i + 1; j < numbers.size(); j++) {
            int target = numbers[i] + numbers[j];
            if (!seen.contains(target)) {
                answer.emplace_back(target);
                seen.insert(target);
            }
        }
    }
    
    sort(answer.begin(), answer.end());

    return answer;
}